### PR TITLE
GpuDebug flag and CSharp glue code

### PR DESCRIPTION
**Pull request by gleblebedev**

### CODE REVIEW

File: Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp [hunk 1]
```diff
@@ -2002,9 +2002,11 @@ bool Graphics::CreateDevice(int width, int height)
     if (!impl_->device_)
     {
         UINT deviceFlags = D3D11_CREATE_DEVICE_BGRA_SUPPORT;
-#ifdef _DEBUG
-        deviceFlags |= D3D11_CREATE_DEVICE_DEBUG;
-#endif
+        if (screenParams_.gpuDebug_)
+        {
+            // Enable the debug layer if requested.
+            deviceFlags |= D3D11_CREATE_DEVICE_DEBUG;
```
Comment (rokups) [inline Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp:2008]: Does OpenGL have a comparable flag?

Comment (gleblebedev) [reply Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp:2008]: I'm not aware of such flag for OpenGL. Vulkan does have debug layer that we can enable in runtime.

The point of the change is to make this switch available in runtime, not in compile time.

**Comment by eugeneko** [comment]
Please either update Window sample (which fails) or make this new parameter optional (meh)

**Comment by gleblebedev** [comment]
> Please either update Window sample (which fails) or make this new parameter optional (meh)
Done. Missed that file

**Pull request closed without merging**